### PR TITLE
Use the correct configuration directory

### PR DESCRIPTION
**Pull request by Delta-official**
updated main.ts so it uses `app.getPath("appData")` instead of `app.getPath("documents")`

### CODE REVIEW

**Comment by Delta-official** [comment]
at least allow linux to use appdata instead of documents, documents is absolutely the wrong place to put stuff like configuration in on linux

**Comment by Mixaz017** [comment]
I agree that configuration files definitely shouldn't be stored in the documents folder. I'm not sure about swapper or script paths though. In Linux way, `~/.config` seems ideal, but in Windows, using documents folder for file based customizations is common.

**Comment by Mixaz017** [comment]
Deciding where to put stuff based on how easy to access them isn't clever imo, ultimately you'll end up putting everything in the desktop folder. Applications should store files based on how organized they are.

**Comment by KraXen72** [comment]
> at least allow linux to use appdata instead of documents, documents is absolutely the wrong place to put stuff like configuration in on linux

i could make it so if there is an appdata folder called Crankshaft, it uses that. Otherwise it falls back to documents. most users already have the settings in documents and it would be a breaking change. This way if you really want to you can use appdata/config on linux.

**Comment by KraXen72** [comment]
if you re-implement it this way, i'll merge it

**Comment by Delta-official** [comment]
ok, i'm gonna implement that

**Comment by Delta-official** [comment]
done

**Comment by KraXen72** [comment]
ok, will review and merge when home. from a quick look on mobile it looks fine.

**Comment by Delta-official** [comment]
that's a long walk to home

**Comment by KraXen72** [comment]
i forgot. i will merge it soon tm when im back from vacation.

File: src/main.ts [hunk 1]
```diff
@@ -11,11 +11,20 @@ import { Swapper } from './resourceswapper';
 // Mixaz and IDKR team - https://github.com/idkr-client/idkr
 // Giant - JANREX client
 // deadcell - css for setting description
-// Tae - logo for the client
+// Tae - logo for>> the client
```
Comment (KraXen72) [inline src/main.ts:14]: revert this, not needed

File: src/main.ts [hunk 2]
```diff
@@ -42,9 +51,17 @@ const settingsSkeleton = {
     experimentalFlags_experimental: false
 }
 
-if (!fs.existsSync(swapperPath)) { fs.mkdirSync(swapperPath, { recursive: true }); };
-if (!fs.existsSync(userscriptsPath)) { fs.mkdirSync(userscriptsPath, { recursive: true }); };
-if (!fs.existsSync(userscriptTrackerPath)) { fs.writeFileSync(userscriptTrackerPath, "{}", {encoding: "utf-8"}) }
+if (!fs.existsSync(swapperPath)) {
```
Comment (KraXen72) [inline src/main.ts:54]: revert this to original, one line state

**Review by KraXen72 — CHANGES_REQUESTED** [review]

**Comment by KraXen72** [comment]
eh i'mma implement this myself

**Comment by KraXen72** [comment]
will give credit dw

**Comment by KraXen72** [comment]
fyi, you don't have to explictly type variables as long as the thing that is filling them with content already has a type / is typed correctly (the fs module has typescript types)
image

**Comment by KraXen72** [comment]
ok so i tried it an there is a pretty major issue:
the appdata directory  for the app does exist on windows by default, since that's where the installer installs it. so on first boot it would make a settings.json there and then the document settings would never get used.

this would mean everybody has to use appdata to configure, which is not what i want. gonna close this for now, as i don't know about a better option. feel free to reopen if you have an idea.

**Pull request closed without merging**